compiler/src/dotty/tools/dotc/core/tasty/PositionPickler.scala

```diff
@@ -19,7 +19,7 @@ import util.Spans._
 class PositionPickler(
     pickler: TastyPickler,
     addrOfTree: PositionPickler.TreeToAddr,
-    treeAnnots: untpd.MemberDef => List[tpd.Tree]) {
+    treeAnnots: untpd.MemberDef => List[tpd.Tree])(using Context) {
 
```

Comment:
We could add the result of `ctx.settings.sourceroot.value` as an argument to avoid adding the context everywhere.

Reply:
Apart form this change the implementation looks good

Reply:
There is still the question if this will work for all the tools. If not it is still a step in the right direction.

Reply:
We had the discussion in #10430 , there we tend to agree with the _single reference point_ scheme in this PR.

Reply:
The issue was not about not creating the field. It was about not mentioning the Context at all in this class. We removed it a while ago to make sure that we only depend on trees and that we could at some point generate an abstract `PositionPickler` that does not depend on the compiler. Having the contexts only invites for missuses of the contexts in this implementation.

```diff
-    treeAnnots: untpd.MemberDef => List[tpd.Tree])(using @constructorOnly ctx: Context) {
+    treeAnnots: untpd.MemberDef => List[tpd.Tree],
+    relativePathReference: String) {
```